# Metagenomics

1. [General introduction](#General-introduction)
2. [Schedule](#Schedule)
    - [Monday](#Monday)
    - [Tuesday](#Tuesday)
    - [Wednesday](#Wednesday)
    - [Thursday](#Thursday)
    - [Friday](#Friday)
3. [Target group](#target-group)
4. [Learning outcomes](#learning-outcomes)
5. [Internet connection](#internet-connection)
6. [Organizers and teachers](#Organizers-and-teachers)


## General introduction
Microbial community structure and diversity, and population structure are fundamental aspects to understand evolution, niche adaptation and demographic history of bacterial species. As NGS sequencing has become cost effective and accessible, sequencing populations of bacteria across the whole genome provides unprecedented resolution to investigate within-host evolution, transmission history, and population structure. Moreover, analysis of genetic content of microbial communities through metagenomics has become the mainstream methodology. During the one week course the students will learn and apply bioinformatic techniques to perform population genetics and study microbial communities with metagenomic approaches. The goal is to become familiar with the bioinformatic analysis tools and to be able to utilize them in own research after the course.

**Everything is possible thanks to the support of CSC**

## Schedule
The course will be held in Viikki at the Faculty of Agriculture and Forestry, University of Helsinki.
Place: Info Centre Korona, room Info4, 1st Floor


### Monday - Viikki C building, lecture room 130
| Time | Description|
| --- | --- |
| 09:00-09:30 | **Jenni Hultman and Mirko Rossi**: Introduction (including general structure of the course and group work)|
| 09:30-09:45 | *Genomics* **Mirko Rossi**: Basic usage of cPouta cloud computing service at CSC |
| 09:45-10:45 | *Genomics* **Mirko Rossi**: [Prepare the Virtual Machine](https://github.com/INNUENDOCON/MicrobialGenomeMetagenomeCourse/blob/master/MPM_starting_VM.md) |
| 10:45-11:00 | *break* - Note: the coffee break is not included in the course |
| 11:00-11:45 | *Genomics* **João Carriço**: Making Use of NGS Data: From Reads to Trees and Annotations |
| 12:00-13:30 | **Working with your data** and **lunch** |
| 13:30-17:00 | *Genomics* **Miguel Machado**: [From reads to assemblies: working with INNUca pipeline](https://github.com/INNUENDOCON/MicrobialGenomeMetagenomeCourse/blob/master/MPM_workingwithINNUCA.md) |

### Tuesday - Biocentre 3, lecture room 1402
| Time | Description |
| --- | --- |
| 08:30-11:45 | *Genomics* **Miguel Machado**: *In silico* typing using ReMatCh and Abricate |
| 12:00-13:30 | **Working with your data** and **lunch** |
| 13:30-16:00 | *Genomics* **Mickael Silva**:  [*Whole genome and core genome MLST analysis using *chewBBACA* suite*](https://github.com/INNUENDOCON/MicrobialGenomeMetagenomeCourse/blob/master/workingwithchewBBACA.md) (see also [*chewBBACA wiki*](https://github.com/B-UMMI/chewBBACA/wiki)) |
| 16:00-17:00 | **Working with your data**

## Wednesday - Viikki C building, lecture room 130
| Time | Description |
| --- | --- |
| 08:30-10:45 | *Genomics* **Miguel Machado & Bruno Gonçalves**: Annotation (*PROKKA*), pangenome anaylysis (*ROARY*) and GWAS (*SCOARY*)|
| 11:00-12:30 | **Working with your data** and **lunch** |
| 12:30-14:30 | *Genomics* **Bruno Gonçalves**: Integrating phylogenetic inference and metadata visualization for NGS data |
| 14:30-15:30 | *Metagenomics* **Antti Karkman**: Data download, QC (*FASTQC*) and trimming (*CUTADAPT*)|
| 15:30-17:00 | *Metagenomics* **Jenni Hultman**: Assembly (*MEGAHIT*) and installation of programs |

## Thusday Lab building, lecture room 228
| Time | Description |
| --- | --- |
| 08:00-10:00 | *Metagenomics* **Jenni Hultman**: Assembly check and mapping with (*BOWTIE2*) |
| 10:00-12:00 | *Metagenomics* **Antti Karkman**: Read based analysis for 16S rRNA (*Metaxa2*) and ARG (*ResFinder*) |
| 12:00-14:00 | **Working with your data** and **lunch** |
| 14:00-17:00 | *Metagenomics* **Antti Karkman and Jenni Hultman**: Prepping files for Anvio |

## Friday Viikki C building, lecture room 130
| Time | Description |
| --- | --- |
| 09:00-10:00 | *Metagenomics* **Tom Delmont**: Introduction to Anvio **NB. Room 2012 Biocenter 2 AND coffee and pulla** |
| 10:30-12:00 | *Metagenomics* **Tom Delmont**: Metagenome visualization in Anvio (*ANVIO*) |
| 12:00-13:00 | **Working with your data** and **lunch** |
| 13:00-16:00 | *Metagenomics* **Tom Delmont**: Construction of MAGs with Anvio |
| 16:00-17:00 | *Metagenomics* **Jenni Hultman**: Wrapping up |

## Target group
This course is targeted at PhD and MSc students who are interested in performing bacterial population analysis using whole-genome sequencing and/or using metagenomics for investigating microbial communities.

## Learning outcomes
QA/QC of the raw sequence data, cleaning and assembly; Gene-by-gene annotation; Visualization of data; Taxonomic classification; Functional assignment, Analyses of microbial community composition; Comparative metagenomics; MAG (MetagenomeAssembledGenome) analysis.

## Internet connection
Inside the class room you can find both __eduroam__ and Univ. Helsinki __HUPnet__. Instructions on how to connect in [HelpDesk website.](https://helpdesk.it.helsinki.fi/en/instructions/logging-and-connections/networks/wireless-connections-university)

## Organizers and Teachers

### Organizers
* Docent Jenni Hultman, University of Helsinki, Finland
* MSc Katariina Pärnänen, University of Helsinki, Finland
* PhD Antti Karkman, University of Helsinki, Finland

### Visiting teachers
* PhD Tom Delmont, XXX, France
* Tommi Vatanen, The University of Auckland, New Zealand
